Stuck in grey screen and reloading dont solve the problem.

**Describe the bug**
Today morning when I opened the desktop app it was stuck in a grey screen. Investigating I saw that it is on a channel and reloading doesn't fix it. There are errors on the console.

What operating system and which version? macOS 12.4
Which version of Rocket.Chat (Server)? Open 5.0
Which version of Rocket.Chat.Electron (Electron/Desktop)? 3.8.7
Is there any setting relevant changed? No

**To Reproduce**
Steps to reproduce the behavior:
No way to reproduce.

**Expected behavior**
We need to add some way to the user go to home, to when a channel breaks like this. Just reloading don't fixed.
On browser is working normally and other client is working well on the server.

**Screenshots**
If applicable, add screenshots to help explain your problem.
![image](https://user-images.githubusercontent.com/1177482/178729781-486b5eac-c036-4b78-a9e8-35108812f964.png)
![image](https://user-images.githubusercontent.com/1177482/178729849-a194bb09-ce04-4c5c-865c-3335dd6cf686.png)


**Additional context**
Add any other context about the problem here.

```
preload.ts:67 [Rocket.Chat Desktop] waiting for window load
preload.ts:26 [Rocket.Chat Desktop] preload.ts start
injected.ts:9 [Rocket.Chat Desktop] Injected.ts
injected.ts:239 [Rocket.Chat Desktop] Injected
injected.ts:12 [Rocket.Chat Desktop] Injected.ts start fired
injected.ts:14 [Rocket.Chat Desktop] window.require is not defined
injected.ts:15 [Rocket.Chat Desktop] Inject start - retrying in 1 seconds
preload.ts:46 [Rocket.Chat Desktop] waiting for RocketChatDesktop.onReady
ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:13 fuselage: 0.31.14
ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:1563 MessageType.render is deprecated. Use MessageType.message instead. livechat_webrtc_video_call
registerType @ ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:1563
injected.ts:12 [Rocket.Chat Desktop] Injected.ts start fired
injected.ts:28 [Rocket.Chat Desktop] Injected.ts serverInfo Object
preload.ts:48 [Rocket.Chat Desktop] RocketChatDesktop.onReady fired
injected.ts:56 [Rocket.Chat Desktop] window.open for Jitsi overloaded 
injected.ts:109 [Rocket.Chat Desktop] Injected.ts replaced Notification
injected.ts:56 [Rocket.Chat Desktop] window.open for Jitsi overloaded meet.jit.si
ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:1127 Error: Cannot find module '../../../../../lib/isE2EEMessage'
    at y (ee6214adb3dafaf539604012652aad7393876aad.js:3:638)
    at d.resolve (ee6214adb3dafaf539604012652aad7393876aad.js:3:3521)
    at d.a [as link] (ee6214adb3dafaf539604012652aad7393876aad.js:5:5456)
    at module (/client/views/room/MessageList/components/MessageContent.tsx:1:567)
    at ee6214adb3dafaf539604012652aad7393876aad.js:1304:764
    at g (ee6214adb3dafaf539604012652aad7393876aad.js:3:1094)
    at d.n [as require] (ee6214adb3dafaf539604012652aad7393876aad.js:3:3652)
    at d.a [as link] (ee6214adb3dafaf539604012652aad7393876aad.js:5:5535)
    at module (/client/views/room/MessageList/components/Message.tsx:1:885)
    at ee6214adb3dafaf539604012652aad7393876aad.js:1304:764
Ti @ ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:1127
ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:1145 Uncaught Error: Cannot find module '../../../../../lib/isE2EEMessage'
    at y (ee6214adb3dafaf539604012652aad7393876aad.js:3:638)
    at d.resolve (ee6214adb3dafaf539604012652aad7393876aad.js:3:3521)
    at d.a [as link] (ee6214adb3dafaf539604012652aad7393876aad.js:5:5456)
    at module (/client/views/room/MessageList/components/MessageContent.tsx:1:567)
    at ee6214adb3dafaf539604012652aad7393876aad.js:1304:764
    at g (ee6214adb3dafaf539604012652aad7393876aad.js:3:1094)
    at d.n [as require] (ee6214adb3dafaf539604012652aad7393876aad.js:3:3652)
    at d.a [as link] (ee6214adb3dafaf539604012652aad7393876aad.js:5:5535)
    at module (/client/views/room/MessageList/components/Message.tsx:1:885)
    at ee6214adb3dafaf539604012652aad7393876aad.js:1304:764
ee6214adb3dafaf539604012652aad7393876aad.js?meteor_js_resource=true:1436 Uncaught (in promise) undefined
```

--- a/src/injected.ts
+++ b/src/injected.ts
@@ -82,6 +82,11 @@ const start = (): void => {
   });
 
   Tracker.autorun(() => {
+    const { gitCommitHash } = Meteor;
+    window.RocketChatDesktop.setGitCommitHash(gitCommitHash);
+  });
+
+  Tracker.autorun(() => {
     const uid = Meteor.userId();
     const isAutoAwayEnabled: unknown = getUserPreference(uid, 'enableAutoAway');
     const idleThreshold: unknown = getUserPreference(uid, 'idleTimeLimit');

--- a/src/injected.ts
+++ b/src/injected.ts
@@ -83,6 +83,7 @@ const start = (): void => {
 
   Tracker.autorun(() => {
     const { gitCommitHash } = Meteor;
+    if (!gitCommitHash) return;
     window.RocketChatDesktop.setGitCommitHash(gitCommitHash);
   });
 

--- a/src/servers/common.ts
+++ b/src/servers/common.ts
@@ -10,6 +10,7 @@ export type Server = {
   lastPath?: string;
   failed?: boolean;
   webContentsId?: number;
+  gitCommitHash?: string;
 };
 
 export const enum ServerUrlResolutionStatus {

--- a/src/servers/main.ts
+++ b/src/servers/main.ts
@@ -7,7 +7,9 @@ import { satisfies, coerce } from 'semver';
 import { invoke } from '../ipc/main';
 import { select, dispatch, listen } from '../store';
 import { hasMeta } from '../store/fsa';
+import { WEBVIEW_GIT_COMMIT_HASH_CHANGED } from '../ui/actions';
 import { getRootWindow } from '../ui/main/rootWindow';
+import { getWebContentsByServerUrl } from '../ui/main/serverView';
 import {
   SERVER_URL_RESOLUTION_REQUESTED,
   SERVER_URL_RESOLVED,
@@ -164,7 +166,22 @@ export const setupServers = async (
     }
   });
 
+  listen(WEBVIEW_GIT_COMMIT_HASH_CHANGED, async (action) => {
+    const { url, gitCommitHash } = action.payload;
+
+    const servers = select(({ servers }) => servers);
+
+    const server = servers.find((server) => server.url === url);
+
+    if (server?.gitCommitHash !== gitCommitHash) {
+      const guestWebContents = getWebContentsByServerUrl(url);
+      await guestWebContents?.session.clearCache();
+      guestWebContents?.reload();
+    }
+  });
+
   let servers = select(({ servers }) => servers);
+
   let currentServerUrl = select(({ currentView }) =>
     typeof currentView === 'object' ? currentView.url : null
   );

--- a/src/servers/preload/api.ts
+++ b/src/servers/preload/api.ts
@@ -6,6 +6,7 @@ import { setUserPresenceDetection } from '../../userPresence/preload';
 import { Server } from '../common';
 import { setBadge } from './badge';
 import { setFavicon } from './favicon';
+import { setGitCommitHash } from './gitCommitHash';
 import { getInternalVideoChatWindowEnabled } from './internalVideoChatWindow';
 import { setBackground } from './sidebar';
 import { setTitle } from './title';
@@ -40,6 +41,7 @@ export type RocketChatDesktopAPI = {
   ) => Promise<unknown>;
   destroyNotification: (id: unknown) => void;
   getInternalVideoChatWindowEnabled: () => boolean;
+  setGitCommitHash: (gitCommitHash: string) => void;
 };
 
 export const RocketChatDesktop: RocketChatDesktopAPI = {
@@ -62,4 +64,5 @@ export const RocketChatDesktop: RocketChatDesktopAPI = {
   createNotification,
   destroyNotification,
   getInternalVideoChatWindowEnabled,
+  setGitCommitHash,
 };

--- a/src/servers/preload/gitCommitHash.ts
+++ b/src/servers/preload/gitCommitHash.ts
@@ -1,0 +1,16 @@
+import { dispatch } from '../../store';
+import { WEBVIEW_GIT_COMMIT_HASH_CHANGED } from '../../ui/actions';
+import { Server } from '../common';
+import { getServerUrl } from './urls';
+
+export const setGitCommitHash = (
+  gitCommitHash: Server['gitCommitHash']
+): void => {
+  dispatch({
+    type: WEBVIEW_GIT_COMMIT_HASH_CHANGED,
+    payload: {
+      url: getServerUrl(),
+      gitCommitHash,
+    },
+  });
+};

--- a/src/servers/reducers.ts
+++ b/src/servers/reducers.ts
@@ -16,6 +16,7 @@ import {
   WEBVIEW_DID_FAIL_LOAD,
   WEBVIEW_READY,
   WEBVIEW_ATTACHED,
+  WEBVIEW_GIT_COMMIT_HASH_CHANGED,
 } from '../ui/actions';
 import { SERVERS_LOADED } from './actions';
 import { Server } from './common';
@@ -36,6 +37,7 @@ type ServersActionTypes =
   | ActionOf<typeof SIDE_BAR_SERVERS_SORTED>
   | ActionOf<typeof WEBVIEW_DID_NAVIGATE>
   | ActionOf<typeof WEBVIEW_SIDEBAR_STYLE_CHANGED>
+  | ActionOf<typeof WEBVIEW_GIT_COMMIT_HASH_CHANGED>
   | ActionOf<typeof WEBVIEW_TITLE_CHANGED>
   | ActionOf<typeof WEBVIEW_UNREAD_CHANGED>
   | ActionOf<typeof WEBVIEW_FAVICON_CHANGED>
@@ -105,6 +107,11 @@ export const servers: Reducer<Server[], ServersActionTypes> = (
     case WEBVIEW_SIDEBAR_STYLE_CHANGED: {
       const { url, style } = action.payload;
       return upsert(state, { url, style });
+    }
+
+    case WEBVIEW_GIT_COMMIT_HASH_CHANGED: {
+      const { url, gitCommitHash } = action.payload;
+      return upsert(state, { url, gitCommitHash });
     }
 
     case WEBVIEW_FAVICON_CHANGED: {

--- a/src/ui/actions.ts
+++ b/src/ui/actions.ts
@@ -94,11 +94,11 @@ export type UiActionTypeToPayloadMap = {
   [SIDE_BAR_SERVER_SELECTED]: Server['url'];
   [SIDE_BAR_SERVERS_SORTED]: Server['url'][];
   [TOUCH_BAR_FORMAT_BUTTON_TOUCHED]:
-  | 'bold'
-  | 'italic'
-  | 'strike'
-  | 'inline_code'
-  | 'multi_line';
+    | 'bold'
+    | 'italic'
+    | 'strike'
+    | 'inline_code'
+    | 'multi_line';
   [TOUCH_BAR_SELECT_SERVER_TOUCHED]: string;
   [UPDATE_DIALOG_DISMISSED]: void;
   [UPDATE_DIALOG_INSTALL_BUTTON_CLICKED]: void;

--- a/src/ui/actions.ts
+++ b/src/ui/actions.ts
@@ -57,6 +57,8 @@ export const WEBVIEW_SCREEN_SHARING_SOURCE_REQUESTED =
 export const WEBVIEW_SCREEN_SHARING_SOURCE_RESPONDED =
   'webview/screen-sharing-source-responded';
 export const WEBVIEW_SIDEBAR_STYLE_CHANGED = 'webview/sidebar-style-changed';
+export const WEBVIEW_GIT_COMMIT_HASH_CHANGED =
+  'webview/git-commit-hash-changed';
 export const WEBVIEW_TITLE_CHANGED = 'webview/title-changed';
 export const WEBVIEW_UNREAD_CHANGED = 'webview/unread-changed';
 export const SETTINGS_SET_REPORT_OPT_IN_CHANGED =
@@ -92,11 +94,11 @@ export type UiActionTypeToPayloadMap = {
   [SIDE_BAR_SERVER_SELECTED]: Server['url'];
   [SIDE_BAR_SERVERS_SORTED]: Server['url'][];
   [TOUCH_BAR_FORMAT_BUTTON_TOUCHED]:
-    | 'bold'
-    | 'italic'
-    | 'strike'
-    | 'inline_code'
-    | 'multi_line';
+  | 'bold'
+  | 'italic'
+  | 'strike'
+  | 'inline_code'
+  | 'multi_line';
   [TOUCH_BAR_SELECT_SERVER_TOUCHED]: string;
   [UPDATE_DIALOG_DISMISSED]: void;
   [UPDATE_DIALOG_INSTALL_BUTTON_CLICKED]: void;
@@ -119,6 +121,10 @@ export type UiActionTypeToPayloadMap = {
   };
   [WEBVIEW_TITLE_CHANGED]: { url: Server['url']; title: Server['title'] };
   [WEBVIEW_UNREAD_CHANGED]: { url: Server['url']; badge: Server['badge'] };
+  [WEBVIEW_GIT_COMMIT_HASH_CHANGED]: {
+    url: Server['url'];
+    gitCommitHash: Server['gitCommitHash'];
+  };
   [SETTINGS_SET_REPORT_OPT_IN_CHANGED]: boolean;
   [SETTINGS_SET_FLASHFRAME_OPT_IN_CHANGED]: boolean;
   [SETTINGS_SET_HARDWARE_ACCELERATION_OPT_IN_CHANGED]: boolean;

